File: examples/typescript/end-users/revokeDelegationForAccount.ts
// Usage: pnpm tsx end-users/revokeDelegationForAccount.ts <USER_UUID> <ADDRESS>

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const userId = process.argv[2];
const address = process.argv[3];

if (!userId || !address) {
  console.error(
    "Usage: pnpm tsx end-users/revokeDelegationForAccount.ts <USER_UUID> <ADDRESS>",
  );
  process.exit(1);
}

const cdp = new CdpClient();

try {
  // Revoke the account-scoped delegation via the client method
  await cdp.endUser.revokeDelegationForEndUserAccount({
    userId,
    address,
  });

  console.log("Revoked account-scoped delegation via client method");

  // Alternatively, use the EndUserAccount object shorthand
  const endUser = await cdp.endUser.getEndUser({ userId });

  await endUser.revokeDelegationForAccount({ address });

  console.log("Revoked account-scoped delegation via account method");
} catch (error) {
  console.error("Error: ", (error as { errorMessage: string }).errorMessage);
}

File: examples/python/end-users/revoke_delegation_for_account.py
"""Example: Revoke the active account-scoped delegation for an end user address.

Usage: python revoke_delegation_for_account.py <USER_UUID> <ADDRESS>
"""

import asyncio
import sys

from cdp import CdpClient


async def main(user_id: str, address: str) -> None:
    """Revoke an account-scoped delegation via client method and account shorthand."""
    async with CdpClient() as cdp:
        # Via the client method
        await cdp.end_user.revoke_delegation_for_end_user_account(
            user_id=user_id,
            address=address,
        )
        print("Revoked delegation via client method")

        # Via the EndUserAccount shorthand
        end_user = await cdp.end_user.get_end_user(user_id=user_id)
        await end_user.revoke_delegation_for_account(address=address)
        print("Revoked delegation via account method")


if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: python revoke_delegation_for_account.py <USER_UUID> <ADDRESS>")
        sys.exit(1)

    asyncio.run(main(sys.argv[1], sys.argv[2]))
